【报Bug】Vue3 下百度 App 使用 NA 渲染小程序时渲染不正常

### 问题描述
Vue3 项目，百度 App 使用 NA 渲染小程序时，渲染特殊组件如 linear-gradient 组件，colors 属性无法生效。

### 复现步骤
**百度 App 中使用 NA 渲染小程序需要命中特殊的实验**，因此需要提供下系统信息、宿主版本、cuid，我这边给该设备命中实验后才能测试该问题。获取方式：
1. 进入百度小程序后，点击小程序右上角的三个点
2. 点击小程序的图标
3. 再连击头像五次，就会显示相关信息

示例：
<img width="439" alt="infoflow 2022-11-10 16-17-44" src="https://user-images.githubusercontent.com/37441990/201037604-703704dd-2a1e-4a47-94db-2ccea2ce2653.png">

####  测试 demo 链接
https://github.com/changhaoliang/uni-app-project

分别 cd demo-vue2, demo-vue3，
`npm i`
`yarn dev:mp-baidu`

用百度小程序开发者工具生成预览码用**真机**扫码预览。

### 实际结果
demo-vue2 目录下的页面 pages/index/index 可以正常渲染渐变色组件，效果如下
<img width="395" alt="infoflow 2022-11-10 15-44-10" src="https://user-images.githubusercontent.com/37441990/201029961-2a40a195-d561-47e6-aa85-2477a6a9cd3c.png">

demo-vue3 目录下渲染不正常，效果如下
<img width="401" alt="infoflow 2022-11-10 15-47-28" src="https://user-images.githubusercontent.com/37441990/201030750-bf218e65-e24c-46d8-b9f1-01220a873c9f.png">

#### 组件名称： linear-gradient
非官方文档提供的组件，是百度小程序使用 NA 渲染时的一个特殊组件。当且仅当 page.json 中配置了 "viewMode": "na" 或 "viewMode": "mini" 时可用。

该组件的属性包括：
- colors
- start
- end
- locations
- useAngle
- angle
- angleCenter
- border-radius

希望如上属性保持原状输入给百度小程序，比如，输入给百度开发者工具的内容应是
```html
<linear-gradient
    style="height: 80px; border-radius: 10px"
    colors="{{['red', 'green', 'black']}}"
    border-radius="{{[10,10, 10,10, 10,10, 10,10]}}">
</linear-gradient>
```

而非类似这样的内容。
```html
<linear-gradient s-if="{{a}}" style="width:300px;height:400px" u-i="d5865606-0" u-p="{{a}}" />
```

### 系统信息

发行平台: 百度小程序
操作系统: 安卓 iOS 双端
uni-app版本：3.0.0-alpha-3050420220804004

为什么编译源码后，在项目代码里安装了改动后的 uni-app-baidu 依赖后，项目编译会报错呢？

Error: Cannot find module '/Users/lchangha/Documents/uni-app/packages/uni-mp-vite/node_modules/@dcloudio/uni-cli-shared/dist/index.js'. Please verify that the package.json has a valid "main" entry

--- a/packages/uni-mp-baidu/src/compiler/options.ts
+++ b/packages/uni-mp-baidu/src/compiler/options.ts
@@ -12,7 +12,25 @@ import { transformFor } from './transforms/vFor'
 import { transformOn } from './transforms/vOn'
 import { transformModel } from './transforms/vModel'
 
-export const customElements = ['follow-swan', 'login', 'inline-payment-panel']
+export const customElements = [
+  'follow-swan',
+  'login',
+  'inline-payment-panel',
+  'talos-linear-gradient',
+  'talos-rc-view',
+  'talos-nested-scroll-view',
+  'talos-nested-scroll-top-container',
+  'talos-nested-scroll-bottom-container',
+  'talos-waterfall-view',
+  'talos-waterfall-item',
+  'talos-waterfall-header',
+  'talos-waterfall-footer',
+  'talos-pull-refresh',
+  'talos-control-container',
+  'talos-na-refresh-control',
+  'talos-modal',
+  'talos-svg',
+]
 const nodeTransforms = [transformRef, transformFor, transformMatchMedia]
 const directiveTransforms = {
   on: transformOn,

